src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -351,8 +356,37 @@ public static bool TryText(FormattingInfo formatInfo, IRContext irContext, Formu
                 return false;
             }
 
-            Contract.Assert(StringValue.AllowedListConvertToString.Contains(value.Type));
-
+            Contract.Assert(StringValue.AllowedListConvertToString.Contains(value.Type));
+
+            bool hasFormatCulture = false;
+            string enUsCultureName = "en-US";
+            CultureInfo formatCulture = culture;
+            string revFormatStr = formatString;
+
+            if (!string.IsNullOrEmpty(formatString) && formatString.StartsWith("[$-", StringComparison.OrdinalIgnoreCase))
+            {
+                int startIdx = formatString.IndexOf("[$-", StringComparison.Ordinal) + 3;
+                int endIdx = formatString.IndexOf("]", StringComparison.Ordinal);
+
+                if (endIdx > 0)
+                {
+                    hasFormatCulture = true;
+                    string formatCultureName = formatString.Substring(startIdx, endIdx - startIdx);
+
+                    if (!TryGetCulture(formatCultureName, out formatCulture))
+                    {
+                        result = new StringValue(irContext, formatCultureName);
+                        return false;
+                    }
+
+                    formatString = formatString.Substring(endIdx + 1);
+                    revFormatStr = formatString.Replace(",", "!");
+                    revFormatStr = revFormatStr.Replace(".", ",");
+                    revFormatStr = revFormatStr.Replace(" ", ",");
+                    revFormatStr = revFormatStr.Replace("!", ".");

```

Comment:
What is this doing?

Reply:
format string in c# only work for en-US format. They don't understand the , and . in different culture. 
As in https://learn.microsoft.com/en-us/dotnet/standard/base-types/custom-numeric-format-strings.
So my idea here is if we pass different culture format, we finding the reverse version of it (will be in en-US format if the input is non en-US and have different format like fr-fr or vi-vi). 
Then we use both format, the input string format and reverse input string format to compare the results of output string when we applied 2 different cultures we have here.
If any of comparison return different mean 2 cultures have different numeric format. Then we need to use reverse format (english format) first then applied that different culture (fr-fr for example).